k3k-kubelet/provider/provider.go

```diff
@@ -177,13 +185,96 @@ func (p *Provider) AttachToContainer(ctx context.Context, namespace, podName, co
 }
 
 // GetStatsSummary gets the stats for the node, including running pods
-func (p *Provider) GetStatsSummary(context.Context) (*statsv1alpha1.Summary, error) {
-	return nil, fmt.Errorf("not implemented")
+func (p *Provider) GetStatsSummary(ctx context.Context) (*statsv1alpha1.Summary, error) {
+	p.logger.Debug("GetStatsSummary")
+
+	nodeList := &v1.NodeList{}
+	if err := p.CoreClient.RESTClient().Get().Resource("nodes").Do(ctx).Into(nodeList); err != nil {
+		return nil, fmt.Errorf("unable to get nodes of cluster %s in namespace %s: %w", p.ClusterName, p.ClusterNamespace, err)
```

Comment:
can we use errors.New() instead of fmt.Errorf, we started to standardize this across all the repo

Reply:
Hm, they are not the same. If you are just printing a string they are equivalent (even though it depends about the number of arguments formatted, and the readability). But if you are formatting an error it's useful to use the `fmt.Errorf` with the `%w` verb. [Since Go 1.13](https://go.dev/blog/go1.13-errors#errors-in-go-113) it wraps the error, and the caller could eventually check for specific errors with the `errors.Is` or `errors.As`.